_posts/2020-10-01-matrix-diagonal-sum.md

```diff
@@ -29,9 +29,14 @@ mat = [
 ## Solution
 
 ```python
-def diagonal_sum(mat):
-
-  return
+def diagonal_sum(A):
+  N = len(A)
+  res = 0
+  for i in range(N):
+      res += A[i][i] + A[i][N - 1 - i]
+  if N & 1:
+      res -= A[N // 2][N // 2]        
+  return res
```

Comment:
Just replacing `A` with `mat`, and adding white space. Also replacing `N` with `n` since all-caps variable names in python are considered to be global.
```suggestion
def diagonal_sum(mat):
  n = len(mat)
  res = 0

  for i in range(n):
    res += mat[i][i] + mat[i][n - 1 - i]

  if n & 1:  # this "bitwise and" operation determines whether n is odd
    res -= mat[n // 2][n // 2]        

  return res
```